types/src/ledger_info.rs

```diff
@@ -376,9 +378,58 @@ impl LedgerInfoWithVerifiedSignatures {
     }
 }
 
-pub enum SignatureWithStatus {
-    Verified(bls12381::Signature),
-    Unverified(bls12381::Signature),
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum VerificationStatus {
+    Verified,
+    Unverified,
+}
+
+#[derive(Clone, Debug, Derivative)]
+#[derivative(PartialEq, Eq)]
+pub struct SignatureWithStatus {
+    signature: bls12381::Signature,
+    #[derivative(PartialEq = "ignore")]
+    status: Arc<RwLock<VerificationStatus>>,
```

Comment:
this can just be a AtomicBool instead of arc of rw lock

Reply:
I tried AtomicBool after your comment. It seems AtomicBool doesn't implement Clone operation. We need Clone for Vote, OrderVote, etc. So, unfortunately, couldn't use AtomicBool.

Reply:
Arc\<AtomicBool\>

Reply:
Done.